producer-c/producer-cloudwatch-integ/KvsVideoOnlyStreamingSampleCloudwatch.cpp

```diff
@@ -0,0 +1,228 @@
+
+        if((retStatus = addFileLoggerPlatformCallbacksProvider(pClientCallbacks,
+                                                               CANARY_FILE_LOGGING_BUFFER_SIZE,
+                                                               CANARY_MAX_NUMBER_OF_LOG_FILES,
+                                                               (PCHAR) FILE_LOGGER_LOG_FILE_DIRECTORY_PATH,
+                                                               TRUE) != STATUS_SUCCESS)) {
+            printf("File logging enable option failed with 0x%08x error code\n", retStatus);
+        }
+        CHK_STATUS(createCanaryStreamCallbacks(&cw, streamName, &pStreamCallbacks, pCanaryStreamCallbacks));
+        CHK_STATUS(addStreamCallbacks(pClientCallbacks, pStreamCallbacks));
+        CHK_STATUS(createKinesisVideoClient(pDeviceInfo, pClientCallbacks, &clientHandle));
+        CHK_STATUS(createKinesisVideoStreamSync(clientHandle, pStreamInfo, &streamHandle));
+
+        // setup dummy frame
+        frame.size = CANARY_METADATA_SIZE + fragmentSizeInByte / DEFAULT_FPS_VALUE;
+        frame.frameData = (PBYTE) MEMALLOC(frame.size);
+        frame.version = FRAME_CURRENT_VERSION;
+        frame.trackId = DEFAULT_VIDEO_TRACK_ID;
+        frame.duration = 0;
+        frame.decodingTs = defaultGetTime(); // current time
+        frame.presentationTs = frame.decodingTs;
+
+        while (ATOMIC_LOAD_BOOL(&sigCaptureInterrupt) != TRUE) {
+            if (frameIndex < 0) {
+                frameIndex = 0;
+            }
+            frame.index = frameIndex;
+            frame.flags = frameIndex % DEFAULT_KEY_FRAME_INTERVAL == 0 ? FRAME_FLAG_KEY_FRAME : FRAME_FLAG_NONE;
+
+            createCanaryFrameData(&frame);
+
+            if (frame.flags == FRAME_FLAG_KEY_FRAME) {
+                if (lastKeyFrameTimestamp != 0) {
+                    CanaryStreamRecordFragmentEndSendTime(pStreamCallbacks, lastKeyFrameTimestamp, frame.presentationTs);
+                    ComputeStreamMetricsFromCanary(streamHandle, pCanaryStreamCallbacks);
+                    ComputeClientMetricsFromCanary(clientHandle, pCanaryStreamCallbacks);
+                    GetCurrentMemoryAllocation(pCanaryStreamCallbacks);
+                }
+                lastKeyFrameTimestamp = frame.presentationTs;
+            }
+            CHK_STATUS(putKinesisVideoFrame(streamHandle, &frame));
+
+            THREAD_SLEEP(HUNDREDS_OF_NANOS_IN_A_SECOND / DEFAULT_FPS_VALUE);
+
+            frame.decodingTs = defaultGetTime(); // current time
+            frame.presentationTs = frame.decodingTs;
+            frameIndex++;
+        }
+
+    }
+CleanUp:
+
+    Aws::ShutdownAPI(options);
+
+    if (STATUS_FAILED(retStatus)) {
+        defaultLogPrint(LOG_LEVEL_ERROR, "", "Failed with status 0x%08x\n", retStatus);
+    }
+
+    if (frame.frameData != NULL) {
+        MEMFREE(frame.frameData);
+    }
+
+    if (pDeviceInfo != NULL) {
+        freeDeviceInfo(&pDeviceInfo);
```

Comment:
These should be idempotent functions so you don't need to check for null

Reply:
Fixed.